// autolesshint/lib/suggesters/attributeQuotesSuggester.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Adds fix suggestions for the attribute_quotes rule.
 */
var AttributeQuotesSuggester = /** @class */ (function () {
    function AttributeQuotesSuggester() {
    }
    /**
     * Suggests a mutation to fix a complaint, if possible.
     *
     * @param complaint   Complaint result from running Lesshint.
     * @returns Suggested mutation for the fix.
     */
    AttributeQuotesSuggester.prototype.suggestMutation = function (complaint) {
        var end = complaint.position + (complaint.source.indexOf("]") - complaint.source.indexOf("="));
        return {
            mutations: [
                {
                    insertion: "'",
                    range: {
                        begin: complaint.position
                    },
                    type: "text-insert"
                },
                {
                    insertion: "'",
                    range: {
                        begin: end
                    },
                    type: "text-insert"
                }
            ],
            range: {
                begin: complaint.position,
                end: end
            },
            type: "multiple"
        };
    };
    return AttributeQuotesSuggester;
}());
exports.AttributeQuotesSuggester = AttributeQuotesSuggester;
