// autolesshint/lib/suggesters/borderZeroSuggester.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Adds fix suggestions for the border_zero rule.
 */
var BorderZeroSuggester = /** @class */ (function () {
    function BorderZeroSuggester() {
    }
    /**
     * Suggests a mutation to fix a complaint, if possible.
     *
     * @param complaint   Complaint result from running Lesshint.
     * @returns Suggested mutation for the fix.
     */
    BorderZeroSuggester.prototype.suggestMutation = function (complaint, config) {
        var insertion;
        var length;
        if (config.style === "zero") {
            insertion = "0";
            length = "none".length;
        }
        else {
            insertion = "none";
            length = "0".length;
        }
        return {
            insertion: insertion,
            range: {
                begin: complaint.position,
                end: complaint.position + length
            },
            type: "text-swap"
        };
    };
    return BorderZeroSuggester;
}());
exports.BorderZeroSuggester = BorderZeroSuggester;
