// autolesshint/lib/suggesters/commentSuggester.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Adds fix suggestions for the comment rule.
 */
var CommentSuggester = /** @class */ (function () {
    function CommentSuggester() {
    }
    /**
     * Suggests a mutation to fix a complaint, if possible.
     *
     * @param complaint   Complaint result from running Lesshint.
     * @param config   Configuration options for the rule.
     * @param fileInfo   Contents of the source file in various forms.
     * @returns Suggested mutation for the fix.
     */
    CommentSuggester.prototype.suggestMutation = function (complaint, config, fileInfo) {
        var originalInput = fileInfo.linesRaw.join("");
        return {
            range: {
                begin: complaint.position,
                end: originalInput.indexOf("*/", complaint.position) + "*/".length
            },
            type: "text-delete"
        };
    };
    return CommentSuggester;
}());
exports.CommentSuggester = CommentSuggester;
