// autolesshint/lib/suggesters/finalNewlineSuggester.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var utils_1 = require("../utils");
/**
 * Adds fix suggestions for the finalNewline rule.
 */
var FinalNewlineSuggester = /** @class */ (function () {
    function FinalNewlineSuggester() {
    }
    /**
     * Suggests a mutation to fix a complaint, if possible.
     *
     * @param complaint   Complaint result from running Lesshint.
     * @param config   Configuration options for the rule.
     * @param fileInfo   Contents of the source file in various forms.
     * @returns Suggested mutation for the fix.
     */
    FinalNewlineSuggester.prototype.suggestMutation = function (complaint, config, fileInfo) {
        var text = fileInfo.text;
        // Fixes for https://github.com/lesshint/lesshint/issues/431
        if (text[text.length - 1] === "\n") {
            return undefined;
        }
        return {
            insertion: utils_1.getLinebreakStyle(fileInfo.linesRaw),
            range: {
                begin: complaint.position
            },
            type: "text-insert"
        };
    };
    return FinalNewlineSuggester;
}());
exports.FinalNewlineSuggester = FinalNewlineSuggester;
