// autolesshint/lib/suggesters/hexNotationSuggester.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Adds fix suggestions for the hex_notation rule.
 */
var HexNotationSuggester = /** @class */ (function () {
    function HexNotationSuggester() {
    }
    /**
     * Suggests a mutation to fix a complaint.
     *
     * @param complaint   Complaint result from running Lesshint.
     * @returns Suggested mutation for the fix.
     */
    HexNotationSuggester.prototype.suggestMutation = function (complaint, config, fileInfo) {
        var original = fileInfo.text.substring(complaint.position, fileInfo.text.substring(complaint.position + 2).search(/\W/i) + complaint.position + 2);
        return {
            insertion: config.style === "uppercase"
                ? original.toUpperCase()
                : original.toLowerCase(),
            range: {
                begin: complaint.position,
                end: complaint.position + original.length
            },
            type: "text-swap"
        };
    };
    return HexNotationSuggester;
}());
exports.HexNotationSuggester = HexNotationSuggester;
