// autolesshint/lib/suggesters/newlineAfterBlockSuggester.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var utils_1 = require("../utils");
/**
 * Adds fix suggestions for the newline_after_block rule.
 */
var NewlineAfterBlockSuggester = /** @class */ (function () {
    function NewlineAfterBlockSuggester() {
    }
    /**
     * Suggests a mutation to fix a complaint, if possible.
     *
     * @param complaint   Complaint result from running Lesshint.
     * @param config   Configuration options for the rule.
     * @param fileInfo   Contents of the source file in various forms.
     * @returns Suggested mutation for the fix.
     */
    NewlineAfterBlockSuggester.prototype.suggestMutation = function (complaint, config, fileInfo) {
        return {
            insertion: utils_1.getLinebreakStyle(fileInfo.linesRaw),
            range: {
                begin: complaint.position
            },
            type: "text-insert"
        };
    };
    return NewlineAfterBlockSuggester;
}());
exports.NewlineAfterBlockSuggester = NewlineAfterBlockSuggester;
