// autolesshint/lib/suggesters/spaceBetweenParensSuggester.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Adds fix suggestions for the space_between_parens rule.
 */
var SpaceBetweenParensSuggester = /** @class */ (function () {
    function SpaceBetweenParensSuggester() {
    }
    /**
     * Suggests a mutation to fix a complaint.
     *
     * @param complaint   Complaint result from running Lesshint.
     * @param config   Configuration options for the rule.
     * @param fileInfo   Contents of the source file in various forms.
     * @returns Suggested mutation for the fix.
     */
    SpaceBetweenParensSuggester.prototype.suggestMutation = function (complaint, config, fileInfo) {
        return config.style === "one_space"
            ? this.suggestOneSpaceMutation(complaint, fileInfo)
            : this.suggestNoSpaceMutation(complaint, fileInfo);
    };
    /**
     * Suggests a mutation to fix a complaint that there should be a space.
     *
     * @param complaint   Complaint result from running Lesshint.
     * @param fileInfo   Contents of the source file in various forms.
     * @returns Suggested mutation for the fix.
     */
    SpaceBetweenParensSuggester.prototype.suggestOneSpaceMutation = function (complaint, fileInfo) {
        return {
            insertion: " ",
            range: {
                begin: complaint.position
            },
            type: "text-insert"
        };
    };
    /**
     * Suggests a mutation to fix a complaint that there should be no space.
     *
     * @param complaint   Complaint result from running Lesshint.
     * @param fileInfo   Contents of the source file in various forms.
     * @returns Suggested mutation for the fix.
     */
    SpaceBetweenParensSuggester.prototype.suggestNoSpaceMutation = function (complaint, fileInfo) {
        var spaceLength = fileInfo.text.substring(complaint.position).search(/\S+/gi);
        return {
            range: {
                begin: complaint.position,
                end: complaint.position + spaceLength
            },
            type: "text-delete"
        };
    };
    return SpaceBetweenParensSuggester;
}());
exports.SpaceBetweenParensSuggester = SpaceBetweenParensSuggester;
