// autolesshint/lib/suggesters/stringQuotesSuggester.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Adds fix suggestions for the StringQuotes rule.
 */
var StringQuotesSuggester = /** @class */ (function () {
    function StringQuotesSuggester() {
    }
    /**
     * Suggests a mutation to fix a complaint, if possible.
     *
     * @param complaint   Complaint result from running Lesshint.
     * @param config   Configuration options for the rule.
     * @returns Suggested mutation for the fix.
     */
    StringQuotesSuggester.prototype.suggestMutation = function (complaint, config) {
        var preferredQuote = config.style === "double" ? '"' : "'";
        var currentQuote = config.style === "double" ? "'" : '"';
        var endOffset = complaint.source.lastIndexOf(currentQuote) - complaint.source.indexOf(currentQuote);
        return {
            mutations: [
                {
                    insertion: preferredQuote,
                    range: {
                        begin: complaint.position,
                        end: complaint.position + 1
                    },
                    type: "text-swap"
                },
                {
                    insertion: preferredQuote,
                    range: {
                        begin: complaint.position + endOffset,
                        end: complaint.position + endOffset + 1
                    },
                    type: "text-swap"
                }
            ],
            range: {
                begin: complaint.position,
                end: complaint.position + endOffset
            },
            type: "multiple"
        };
    };
    return StringQuotesSuggester;
}());
exports.StringQuotesSuggester = StringQuotesSuggester;
