// autolesshint/lib/suggesters/urlQuotesSuggester.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Adds fix suggestions for the urlQuotes rule.
 */
var UrlQuotesSuggester = /** @class */ (function () {
    function UrlQuotesSuggester() {
    }
    /**
     * Suggests a mutation to fix a complaint, if possible.
     *
     * @param complaint   Complaint result from running Lesshint.
     * @returns Suggested mutation for the fix.
     */
    UrlQuotesSuggester.prototype.suggestMutation = function (complaint) {
        var end = complaint.position + complaint.source.indexOf(")", complaint.column) - complaint.column + 2;
        return {
            mutations: [
                {
                    insertion: "'",
                    range: {
                        begin: complaint.position
                    },
                    type: "text-insert"
                },
                {
                    insertion: "'",
                    range: {
                        begin: end
                    },
                    type: "text-insert"
                }
            ],
            range: {
                begin: complaint.position,
                end: end
            },
            type: "multiple"
        };
    };
    return UrlQuotesSuggester;
}());
exports.UrlQuotesSuggester = UrlQuotesSuggester;
